src/fontra/views/editor/panel-reference-font.js

```diff
@@ -649,6 +661,44 @@ export default class ReferenceFontPanel extends Panel {
       []
     );
 
+    const addRemoveButtons = createDomElement("add-remove-buttons");
+    addRemoveButtons.addButtonCallback = async () => {
+      // FIXME Add dialog title translation
+      const dialog = await dialogSetup("Upload", null, [
+        {
+          title: translate("dialog.cancel"),
+          resultValue: "cancel",
+          isCancelButton: true,
+        },
+        {
+          title: translate("dialog.add"),
+          resultValue: "ok",
+          isDefaultButton: true,
+        },
+      ]);
+      const fileInput = input({ type: "file", id: "reference-font-file" }, []);
+      dialog.setContent(
+        form({ enctype: "multipart/form-data", class: "content" }, [
+          label({ for: "reference-font-file", class: "reference-font-label" }, [
+            "Choose a file",
+          ]),
+          fileInput,
+        ])
+      );
+      const result = await dialog.run();
+      const { files } = fileInput;
+      if (result === "ok" && files.length === 1) {
+        await this._filesHandler(files);
+      }
+    };
+
+    addRemoveButtons.removeButtonCallback = () => {
+      const index = this.filesUIList.getSelectedItemIndex();
+      if (!isNaN(index) && index <= this.filesUIList.items.length - 1) {
+        this._deleteItemOrAll(index);
+      }
+    };
+
```

Comment:
If there is no reference font, please disable the remove button via `disableRemoveButton`.